akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventWriter.scala

```diff
@@ -273,48 +422,32 @@ private[akka] object EventWriter {
               case _          => repr
             }
 
-            val newStateForPid =
-              perPidWriteState.get(persistenceId) match {
-                case None =>
-                  if (context.log.isTraceEnabled)
-                    context.log.traceN(
-                      "Writing event persistence id [{}], sequence nr [{}], payload {}",
-                      persistenceId,
-                      sequenceNumber,
-                      event)
-                  sendToJournal(1, Vector(reprWithMeta))
-                  StateForPid(
-                    Map((reprWithMeta.sequenceNr, (reprWithMeta, replyTo))),
-                    emptyWaitingForWrite,
-                    currentTransactionId = 1)
-                case Some(state) =>
-                  // write in progress for pid, add write to batch and perform once current write completes
-                  if (state.waitingForWrite.size == eventWriterSettings.maxBatchSize) {
-                    replyTo ! StatusReply.error(
-                      s"Max batch reached for pid $persistenceId, at most ${eventWriterSettings.maxBatchSize} writes for " +
-                      "the same pid may be in flight at the same time")
-                    state
-                  } else {
-                    if (context.log.isTraceEnabled)
-                      context.log.traceN(
-                        "Writing event in progress for persistence id [{}], adding sequence nr [{}], payload {} to batch",
-                        persistenceId,
-                        sequenceNumber,
-                        event)
-                    state.copy(waitingForWrite = state.waitingForWrite :+ ((reprWithMeta, replyTo)))
-                  }
-              }
-            perPidWriteState = perPidWriteState.updated(persistenceId, newStateForPid)
+            handleWrite(reprWithMeta, replyTo)
+            Behaviors.same
+
+          case MaxSeqNrForPid(pid, maxSeqNr, None) =>
+            perPidWriteState.get(pid) match {
+              case None =>
+                context.log.debug("Got max seq nr with no waiting previous state for pid, ignoring (pid [{}])", pid)
+              case Some(state) =>
+                context.log.debug("Retrieved max seq nr [{}] for pid [{}]", maxSeqNr, pid)
+                val waiting = state.waitingForSeqNrLookup
+                perPidWriteState = perPidWriteState.updated(
+                  pid,
+                  state
+                    .copy(latestSeqNr = maxSeqNr, seqNrlookupInProgress = false, waitingForSeqNrLookup = Vector.empty))
+                waiting.foreach { case (repr, replyTo) => handleWrite(repr, replyTo) }
```

Comment:
I thought first that we could perhaps batch those into one write (or a few) if a bunch has been queued up, but I realize there could be gaps _between_ the waitingForSeqNrLookup ones as well, so maybe not worth it.